shim/src/evaluators.js

```diff
@@ -146,10 +146,9 @@ export function createSafeEvaluatorFactory(unsafeRec, safeGlobal) {
           throw e;
         } finally {
           // belt and suspenders: the proxy switches this off immediately after
-          // the first access, but just in case we clear it here too
-          if (scopeHandler.useUnsafeEvaluator !== false) {
-            scopeHandler.useUnsafeEvaluator = false;
-            throwTantrum('handler sets useUnsafeEvaluator = false', err);
+          // the first access, but if that's not the case we abort.
+          if (scopeHandler.hasUnsafeEvaluator()) {
```

Comment:
`hasUnsafeEvaluator` isn't quite right, because the handler possesses it regardless. How about `willEvaluateUnsafely` or `unsafeEvaluatorAllowed` ?